### PR TITLE
Port factorytest

### PR DESCRIPTION
**Pull request by Brycey92**
### Description

Ported factory test mode, modified it for new buzzers and touchpad.

### Test Instructions

Tested on emulator and firmware
- with fresh NVS: mode entered OK
- after a fail-and-reset: mode entered OK
- after a pass-and-reset: mode exited OK, not entered again OK
- after a menu-opened fail-and-reset: mode not entered again OK
- after a menu-opened pass-and-reset: mode not entered again OK

If any of these behaviors are incorrect, I can change them.

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

File: main/CMakeLists.txt
Comment (AEFeinstein) [inline main/CMakeLists.txt:1]: mmm alphabetization

File: main/modes/factoryTest/factoryTest.c [hunk 1]
```diff
@@ -0,0 +1,804 @@
...
+            }
+        }
+    }
+}
+
+/**
+ * @brief Read, save, and validate accelerometer data
+ *
+ */
+void testReadAndValidateAccelerometer(void)
```
Comment (AEFeinstein) [inline main/modes/factoryTest/factoryTest.c:707]: @cnlohr after this gets merged to main and you want to add your motion and calibration stuff, this is likely the right place to do it

File: main/modes/factoryTest/factoryTest.c [hunk 2]
```diff
@@ -0,0 +1,804 @@
...
+
+    // Plot Z accel
+    barWidth = ((test->z + 256) * MAX_ACCEL_BAR_W) / 512;
+    if (barWidth >= 0)
+    {
+        fillDisplayArea(TFT_WIDTH - barWidth, barY, TFT_WIDTH, barY + ACCEL_BAR_HEIGHT, accelColor);
+    }
+    barY += (ACCEL_BAR_HEIGHT + ACCEL_BAR_SEP);
+
+    // Draw the 8-direction touchpad with center circle
```
Comment (AEFeinstein) [inline main/modes/factoryTest/factoryTest.c:313]: I haven't ran this (at work) but I have a preference for drawing the analog dot-in-a-circle. It gives a better sense that analog touch is working as expected.

File: main/modes/mainMenu/mainMenu.c [hunk 3]
```diff
@@ -146,6 +147,7 @@ static void mainMenuEnterMode(void)
     addSingleItemToMenu(mainMenu->menu, accelTestMode.modeName);
     addSingleItemToMenu(mainMenu->menu, demoMode.modeName);
     addSingleItemToMenu(mainMenu->menu, touchTestMode.modeName);
+    addSingleItemToMenu(mainMenu->menu, factoryTestMode.modeName);
```
Comment (AEFeinstein) [inline main/modes/mainMenu/mainMenu.c:150]: At first glance, this shouldn't be added to the main menu. But it's in the test section, which will all get axed anyway.

File: main/modes/touchTest/touchTest.h [hunk 4]
```diff
@@ -5,4 +5,91 @@
...
+ *
+ * @param x X coordinate on the TFT for the center of the circle
+ * @param y Y coordinate on the TFT for the center of the circle
+ * @param r Radius of the circle
+ * @param segs Number of directional segments to split the touchpad into
+ * @param center True to include a center zone, false to only use directional segments
+ * @param touched Whether the touchpad is currently being touched
+ * @param val The output you received from a prior call to `getTouchJoystickZones`
+ */
+static void touchDrawCircle(font_t* font, const char* label, int16_t x, int16_t y, int16_t r, int16_t segs, bool center,
```
Comment (AEFeinstein) [inline main/modes/touchTest/touchTest.h:19]: No no no no. Do not put functions in headers. Ever. Well, except for `inline` but that's special. Each time a header is included it's going to create a new copy of this function in source code. You made them `static` so they won't collide, but this is awful for code space.

**Review by AEFeinstein — APPROVED** [review]

File: main/modes/mainMenu/mainMenu.c (hunk 3 above)
Comment (Brycey92) [reply main/modes/mainMenu/mainMenu.c:150]: Easter egg'd you mean

File: main/modes/touchTest/touchTest.h (hunk 4 above)
Comment (Brycey92) [reply main/modes/touchTest/touchTest.h]: Oh, I thought that was the case but it HATED when I tried to declare the function in the header and define it in the c file, `extern` or not

Comment (Brycey92) [reply main/modes/touchTest/touchTest.h]: Ah, looks like `static` was my problem. Boy did Java set me up for failure when it comes to that keyword; I can never remember which purpose it serves in which language 😭

Comment (AEFeinstein) [reply main/modes/touchTest/touchTest.h]: Yeah, static is weird in C. A static function can only exist in it's own file. A static variable outside a function is the same. A static variable in a function persists it's value between function calls. Be careful

**Pull request merged**